src/rnnlm/rnnlm-core-training.cc

```diff
@@ -257,7 +337,8 @@ void RnnlmCoreTrainer::ProcessOutput(
                      &weight, &objf_num, &objf_den,
                      &objf_den_exact);
 
-  objf_info_.AddStats(weight, objf_num, objf_den, objf_den_exact);
+  if (is_backstitch_step1)
+    objf_info_.AddStats(weight, objf_num, objf_den, objf_den_exact);
```

Comment:
Is there a reason why we are only doing this for the back step?

Reply:
Doing this for the back step corresponds to the stats computed from the parameters updated   after the whole 2-step update on a minibatch